pyharp/device.py

```diff
@@ -241,67 +242,91 @@ def enable_alive_en(self):
+    #         if self._ser.inWaiting():
+    #             break
+    #         if perf_counter() - read_start >= self.__class__.TIMEOUT_S:
+    #             break
+    #     try:
+    #         message_type = self._ser.read(1)[0]  # byte array with only one byte
+    #         message_length = self._ser.read(1)[0]
+    #         message_content = self._ser.read(message_length)
+    #         self.log.debug(f"reply (type): {message_type}")
+    #         self.log.debug(f"reply (length): {message_length}")
+    #         self.log.debug(f"reply (payload): {message_content}")
+    #         # print(f"Read back:")
+    #         # print(f"  type: {MessageType(message_type).name}")
+    #         # print(f"  length : {repr(message_length)}")
+    #         # print(f"  payload: {list(message_content)}")
+
+    #         frame = bytearray()
+    #         frame.append(message_type)
+    #         frame.append(message_length)
+    #         frame += message_content
+    #         msg = HarpMessage.parse(frame)
+
+    #         return msg
+    #     except IndexError:
+    #         return None
 
     def _read(self) -> Union[ReplyHarpMessage, None]:
         """(Blocking) Read an incoming serial message."""
-        # block for up to TIMEOUT until we get at least one byte.
-        read_start = perf_counter()
-        while True:
-            if self._ser.inWaiting():
-                break
-            if perf_counter() - read_start >= self.__class__.TIMEOUT_S:
-                break
         try:
-            message_type = self._ser.read(1)[0]  # byte array with only one byte
-            message_length = self._ser.read(1)[0]
-            message_content = self._ser.read(message_length)
-            self.log.debug(f"reply (type): {message_type}")
-            self.log.debug(f"reply (length): {message_length}")
-            self.log.debug(f"reply (payload): {message_content}")
-            #print(f"Read back:")
-            #print(f"  type: {MessageType(message_type).name}")
-            #print(f"  length : {repr(message_length)}")
-            #print(f"  payload: {list(message_content)}")
-
-            frame = bytearray()
-            frame.append(message_type)
-            frame.append(message_length)
-            frame += message_content
-            msg = HarpMessage.parse(frame)
-
-            return msg
-        except IndexError:
+            return self._ser.msg_q.get(block=True, timeout=self.read_timeout_s)
+        except queue.Empty:
             return None
 
-
     def _dump_reply(self, reply: bytes):
         assert self._dump_file_path is not None
         with self._dump_file_path.open(mode="ab") as f:
             f.write(reply)
+
+    # def read_register(self, register_name: str):
+    #     register: Register = CommonRegisters[register_name]
+    #     ReadHarpMessage(register.type, register.address)
+
+    # def write_register(self, register_name: str, value):
+
+    def get_events(self) -> list[ReplyHarpMessage]:
```

Comment:
This looks great! Let's add:
1. a docstring
2. another function called `event_count()` or `num_events()` that returns the number of event messages in the queue.